build/Build.md

```diff
@@ -0,0 +1,190 @@
+
+    + *The product distribution pack and its dependencies are removed to prevent the Docker image size from increasing. <br/> Only the necessary software packages have been installed.* (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#dont-install-unnecessary-packages) for more details)
+
+* *Non-root user:* 
+
+    + *WSO2-released Docker images ship with ‘wso2carbon’ non-root user for running WSO2 products.*
+
+* *Heightened Clarity:* 
+
+    + *Due to sorted multi-line arguments.* (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments) for more details)
+
+In case of an extension, WSO2 recommends using an official WSO2 product Docker image as the base of that extended Docker image. 
+
+<br/>
+
+## 1.3. Building from source
+
+In case if the preference is to build the images from source, since this gives you greater control over the image and the conditions <br/>affecting its security, WSO2 recommends referring to the official GitHub release of the Docker resources of the relevant <br/>WSO2 product to write your dockerfile and build a WSO2 product Docker image by source.
+
+Also in case if your preferred JDK is different from the default that we pack, you can follow this approach to build your own docker image.
+
+Useful reference points:
+
+[1] [Docker resources for WSO2 API Management](https://github.com/wso2/docker-apim) <br/>
+[2] [Docker resources for WSO2 Integration](https://github.com/wso2/docker-ei) <br/>
+[3] [Docker resources for WSO2 Identity and Access Management](https://github.com/wso2/docker-is)
+
+<br/>
+
+## 1.4. Tips to harden your image
+
+Key takeaways from our experience to write your dockerfile :
+
+```
+[1] Verify Authenticity of any software installed in the image
+[2] Always use HTTPS for files added remotely
+[3] From directive: Use specific / unique tag
+[4] Never run as root
+[5] USER directive: Always
+[6] Drop privileges where necessary
+[7] Expose only required ports
+```
+
+Useful reference points:
+
+[1] [Docker resources for WSO2 API Management](https://github.com/wso2/docker-apim) <br/>
+[2] [Docker resources for WSO2 Integration](https://github.com/wso2/docker-ei) <br/>
+[3] [Docker resources for WSO2 Identity and Access Management](https://github.com/wso2/docker-is)
+
+<br/>
+
+## 1.5. Tips on Tagging
+
+WSO2 product docker image tagging strategy is simple.
+
+[1] Per each product version, we maintain an image queue. <br/>
+This queue can hold a maximum of 3 images from the oldest to the latest and each bear a unique tag to distinguish itself from the other.
+
+* *E.g.* 	
+        <br/>
+        *wso2am:3.1.0.1591023165861.2 - Oldest* <br/>
```

Comment:
@DilanUA just to note, we will be removing the Docker release tag version number from the image tag and moving it to image metadata (e.g., see https://github.com/wso2/docker-apim/issues/338).

Also, we may have to adjust the examples as per the new Updates story.

Reply:
Let's change once the changes are reflected in the registry.